### DISCUSSION TITLE
Custom toReponse that is not JSON

### DISCUSSION THREAD

**Discussion by iamrgroot** [spatie/laravel-data#436] [Q&A] [answered]
I'm overwriting the toResponse function of my Data class. However, since we have to return XML instead of JSON I want to return a Response instead of a JsonReponse. This just works

```php
    /**
     * @param Request $request
     *
     * @return Response
     */
    public function toResponse($request): Response
    {
        return new Response(
            content: ArrayToXml::convert(
                array: $this->toArray(),
            ),
            headers: [
                'Content-Type' => 'application/xml',
            ]
        );
    }
```

However, we run into the phpstan error about the types: 
> Return type (Illuminate\Http\Response) of method App\Data\Test::toResponse() should be compatible with return type (Illuminate\Http\JsonResponse) of method Spatie\LaravelData\Data::toResponse()

Is there any what I can fix the error without a phpstan-ignore?

If not, would it be an option to change the PHPdoc return type of `Spatie\LaravelData\Data::toResponse()` to `Illuminate\Http\Response` in stead of `Illuminate\Http\JsonResponse`?

**Comment by rubenvanassche** [comment, marked as answer]
I think you'll need to return a `Symfony\Component\HttpFoundation\Response` to both capture `Illuminate\Http\Respons` and `Illuminate\Http\JsonResponse`. I'll merge the PR if you send it in!

**Comment by iamrgroot** [comment]
See #437
